addons/l10n_es_edi_tbai/models/account_edi_format.py

```diff
@@ -0,0 +1,406 @@
+            values['regime_key'].append(52)  # code for simplified invoices
+        elif any(filter(
+                lambda l: any(filter(
+                    lambda d: d.get('TipoRecargoEquivalencia'),
+                    l.get('Sujeta', {}).get('NoExenta', {}).get('DesgloseIVA', {}).get('DetalleIVA', {})
+                )),
+                (inv_details, inv_details.get("PrestacionServicios", {}), inv_details.get("Entrega", {}))
+        )):
+            values['regime_key'].append(51)  # code for invoices with "recargo" taxes TODO wrong condition ?
+
+        return values
+
+    def _l10n_es_tbai_get_trail_values(self, invoice, cancel):
+        prev_invoice = invoice.company_id.get_l10n_es_tbai_last_posted_invoice()
+        if prev_invoice and not cancel:
+            return {
+                'chain_prev_invoice': prev_invoice
+            }
+        else:
+            return {}
+
+    def _l10n_es_tbai_sign_invoice(self, invoice, xml_root):
+        company = invoice.company_id
+        cert_private, cert_public = company.l10n_es_tbai_certificate_id._get_key_pair()
+        public_key = cert_public.public_key()
+
+        p12 = company.l10n_es_tbai_certificate_id._get_p12()
+        cert_p12 = p12.get_certificate()
+        issuer = cert_p12.get_issuer()
+
+        # Identifiers
+        document_id = "Document-" + str(uuid4())
+        signature_id = "Signature-" + document_id
+        keyinfo_id = "KeyInfo-" + document_id
+        sigproperties_id = "SignatureProperties-" + document_id
+
+        # Render digital signature scaffold from QWeb
+        web_services = get_web_services_for_agency(agency=company.l10n_es_tbai_tax_agency, env=self.env)
+        values = {
+            'dsig': {
+                'document_id': document_id,
+                'x509_certificate': L10nEsTbaiXmlUtils._base64_print(b64encode(cert_public.public_bytes(encoding=serialization.Encoding.DER))),
+                'public_modulus': L10nEsTbaiXmlUtils._base64_print(b64encode(L10nEsTbaiXmlUtils._int_to_bytes(public_key.public_numbers().n))),
+                'public_exponent': L10nEsTbaiXmlUtils._base64_print(b64encode(L10nEsTbaiXmlUtils._int_to_bytes(public_key.public_numbers().e))),
+                'iso_now': datetime.now().isoformat(),
+                'keyinfo_id': keyinfo_id,
+                'signature_id': signature_id,
+                'sigproperties_id': sigproperties_id,
+                'reference_uri': "Reference-" + document_id,
+                'sigpolicy_description': "Política de Firma TicketBAI 1.0",  # í = &#237;
+                'sigpolicy_url': web_services.get_sigpolicy_url(),
+                'sigpolicy_digest': web_services.get_sigpolicy_digest(),
+                'sigcertif_digest': b64encode(cert_public.fingerprint(hashes.SHA256())).decode(),
+                'x509_issuer_description': f"CN={issuer.CN}, OU={issuer.OU}, O={issuer.O}, C={issuer.C}",
+                'x509_serial_number': cert_p12.get_serial_number(),
```

Comment:
```suggestion
                'x509_issuer_description': ...,
                'x509_serial_number': cert_public.serial_number,
```
Issuer data can be recreated using cert_public.issuer.rdns (see [this comment](https://github.com/odoo/odoo/pull/83883/files#diff-8902cfd630f1d73c831fb20d5567b9b2f2566e9357296a44ec12b23ff8c3ab07R33-R35)). See how the package xmlsig.utils.get_rdns_name() function do it as a ref